src/freedv_interface.cpp

```diff
@@ -15,14 +36,137 @@ FreeDVInterface::~FreeDVInterface()
     if (isRunning()) stop();
 }
 
+
+static void callback_err_fn(void *fifo, short error_pattern[], int sz_error_pattern)
+{
+    codec2_fifo_write((struct FIFO*)fifo, error_pattern, sz_error_pattern);
+}
+
+void FreeDVInterface::start(int txMode, int fifoSizeMs)
+{
+    int src_error = 0;
+    for (auto& mode : enabledModes_)
+    {
+        struct freedv* dv = nullptr;
+        if ((mode == FREEDV_MODE_700D) || (mode == FREEDV_MODE_700E) || (mode == FREEDV_MODE_2020)) {
+            // 700 has some init time stuff so treat it special
+            struct freedv_advanced adv;
+            dv = freedv_open_advanced(mode, &adv);
+        } else {
+            dv = freedv_open(mode);
+        }
+        assert(dv != nullptr);
+        
+        dvObjects_.push_back(dv);
+        
+        struct FIFO* errFifo = codec2_fifo_create(2*freedv_get_sz_error_pattern(dv) + 1);
+        assert(errFifo != nullptr);
+        
+        errorFifos_.push_back(errFifo);
+        
+        auto tmpPtr = new COMP();
+        tmpPtr->real = cos(0.0);
+        tmpPtr->imag = sin(0.0);
+        txFreqOffsetPhaseRectObjs_.push_back(tmpPtr);
+        
```

Comment:
Not sure if you need mode specific tx phase variables, as you're only tx-ing one channel at a time, and the state get's reset at the start of every PTT "over".

Reply:
Yep, looks like a single one is fine. 👍 I consolidated the multiple ones for TX into one.